reasoning_gym/algebra/polynomial_multiplication.py

```diff
@@ -0,0 +1,142 @@
+    single_variable: bool = True,
+    operators: Tuple[str, ...] = (
+        "+",
+        "-",
+    )  # Allowed operators between terms, Avoid adding '*' or '/' because they will affect the degree
+    seed: Optional[int] = None
+    size: int = 500
+
+    def validate(self) -> None:
+        """Validate configuration parameters."""
+        assert self.min_terms > 0, "min_terms must be positive."
+        assert self.max_terms >= self.min_terms, "max_terms must be >= min_terms."
+
+        assert self.min_value > 0, "min_value must be positive."
+        assert self.max_value >= self.min_value, "max_value must be >= min_value."
+
+        assert self.min_degree >= 1, "min_degree must be >= 1."
+        assert self.max_degree >= self.min_degree, "max_degree must be >= min_degree."
+
+        assert self.min_polynomials >= 2, "min_polynomials must be >= 2."
+        assert self.max_polynomials >= self.min_polynomials, "max_polynomials must be >= min_polynomials."
+
+        allowed_ops = {"+", "-"}
+        assert len(self.operators) > 0, "operators tuple cannot be empty."
+        assert all(op in allowed_ops for op in self.operators), "Invalid operator found. Must be a subset of {+, -}."
+
+
+class PolynomialMultiplicationDataset(ProceduralDataset):
+    """
+    Generates [min_polynomials, max_polynomials] random polynomials of degree in [min_degree, max_degree].
+    - The polynomial is formed by summing random terms of the form: coeff * x^exponent.
+    - Then we find "F = P_0 * ... * P_1" using Sympy.
+    """
+
+    def __init__(self, config: PolynomialMultiplicationConfig):
+        self._prompt_templates = [
+            "Simplify this expression: {polynomial_expr}",
+            "Calculate the following: {polynomial_expr}"
+        ]
+        super().__init__(config=config, seed=config.seed, size=config.size)
+
+    def __getitem__(self, idx: int) -> dict:
+        """
+        Generate a single polynomial multiplication item.
+
+        Returns:
+            A dict with:
+                - question: str (e.g. "Multiply polynomials: (8x^3 + x + 2)*(x - 3)")
+                - answer: str (Product, e.g. "8x^4 - 24x^3 + x^2 - x - 6")
+                - metadata: dict with details (polynomial_expr, single_variable)
+        """
+        rng = random.Random(self.seed + idx)
+        number_polynomials = rng.randint(self.config.min_polynomials, self.config.max_polynomials)
+        polynomials = [self._generate_polynomial_expr(rng) for i in range(number_polynomials)]
+
+        polynomial_expr = sp.prod(polynomials)
+        product = sp.expand(polynomial_expr)
+
+        return {
+            "question": rng.choice(self._prompt_templates).format(
+                polynomial_expr=polynomial_expr,
+            ),
+            "answer": product,
+            "metadata": {
+                "polynomial_expr": str(polynomial_expr),
+                "single_variable": self.config.single_variable
+            },
+        }
+
```

Comment:
It would be super-nice if you could add a `score_answer()` function which uses sympy to parse and compare the answer, so that we tolerate whitespaces and different order of terms (e.g. compare by mathematical equality instead exact representation).